### PR TITLE
[feat] Tutorial

### PR DESCRIPTION
**Pull request by ghost**
Stabile Nummer auf Basic Niveau xd
giphy

### CODE REVIEW

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package github.weichware10.toolbox.gui;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+import github.weichware10.toolbox.codecharts.CodeCharts;
+import github.weichware10.toolbox.zoommaps.ZoomMaps;
+import github.weichware10.util.Files;
+import github.weichware10.util.Logger;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:10]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'github.weichware10.util.Logger' import. Should be before 'java.util.ArrayList'.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.text.Text;
+import javafx.stage.Stage;
+
+public class Tutorial {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:22]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    private final Stage primaryStage;
+    private final ConfigClient configClient;
+    private final DataBaseClient dataBaseClient;
+    private TutorialController controller;
+    private static ArrayList<Image> imageList = new ArrayList<>();
+    private final String toolType;
+
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:29]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line has trailing whitespaces.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+public class Tutorial {
+    private final Stage primaryStage;
+    private final ConfigClient configClient;
+    private final DataBaseClient dataBaseClient;
+    private TutorialController controller;
+    private static ArrayList<Image> imageList = new ArrayList<>();
+    private final String toolType;
+    
+    public Tutorial(Stage primaryStage, String toolType,
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:30]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        boolean finished = false;
+        int index = 0;
+        while (!finished) {
+            try {
+                String imageLoc = Files.saveImage(String.format("https://raw.githubusercontent.com/weichware10/dokumente/main/tutorial/%s/%d.png", toolType, index));
+                imageList.add(new Image(Paths.get(imageLoc).toUri().toString()));
+            } catch (IllegalArgumentException | IOException e) {
+                finished = true;
+            }
+        index += 1;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:62]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'while' child has incorrect indentation level 8, expected level should be 12.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        }
+        if(imageList.size() > 0) {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:64]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:64]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            controller.getImageView().setImage(imageList.get(0));
+            controller.getProgressText().setText(String.format("1/%d", imageList.size()));
+        }
+
+        Scene scene = new Scene(root, 300, 275);
+        primaryStage.setScene(scene);
+    }
+
+    public static void setImage(int tutorialCount, Button nextButton, Button backButton, ImageView imageView, Text progressText){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:73]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 129).

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:73]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:73]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        //setzt die Sichtbarkeit des zurück Button
+        if(tutorialCount <= 0) {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:75]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:75]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+            backButton.setDisable(true);
+        }
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:77]: ⚠️ **[checkstyle]** <RightCurlySame> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        else if(tutorialCount > 0) {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:78]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:78]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+            backButton.setDisable(false);
+        }
+        //setzt die Sichtbarkeit des weiter Button
+        if(tutorialCount >= imageList.size()-1) {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '-' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '-' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+            nextButton.setDisable(true);
+        }
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:84]: ⚠️ **[checkstyle]** <RightCurlySame> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).

File: src/main/java/github/weichware10/toolbox/gui/Tutorial.java [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+        else if(tutorialCount < imageList.size()-1) {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:85]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:85]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:85]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '-' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Tutorial.java:85]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '-' is not preceded with whitespace.

**Review by weichbot10 — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>checkstyle</summary>

[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|89 col 66|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L89) WhitespaceAround: '+' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|89 col 66|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L89) WhitespaceAround: '+' is not preceded with whitespace.
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|96 col 5|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L96) Missing a Javadoc comment.
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|97 col 9|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L97) 'if' is not followed by whitespace.
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|97 col 9|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L97) WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|99 col 9|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L99) '}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|100 col 14|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L100) 'if' is not followed by whitespace.
[src/main/java/github/weichware10/toolbox/gui/Tutorial.java|100 col 14|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/Tutorial.java#L100) WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|8 col 1|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L8) Wrong lexicographical order for 'javafx.scene.image.ImageView' import. Should be before 'javafx.scene.layout.HBox'.
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|11 col 1|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L11) Missing a Javadoc comment.
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|25 col 18|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L25) Member name 'hBoxId' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$' (camelCase, start with lowercase, only alphanumerical).
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|74|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L74) Line is longer than 100 characters (found 117).
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|75|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L75) Line is longer than 100 characters (found 107).
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|76|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L76) Line is longer than 100 characters (found 119).
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|77|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L77) Line is longer than 100 characters (found 119).
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|78|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L78) Line is longer than 100 characters (found 123).
[src/main/java/github/weichware10/toolbox/gui/TutorialController.java|79|](http://github.com/weichware10/toolbox/blob/d292d675f114bd4acd02f6263c61063f4e6147b5/src/main/java/github/weichware10/toolbox/gui/TutorialController.java#L79) Line is longer than 100 characters (found 123).
</details>

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
+package github.weichware10.toolbox.gui;
+
+import java.net.URL;
+import java.util.ResourceBundle;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.layout.HBox;
+import javafx.scene.image.ImageView;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:8]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.scene.image.ImageView' import. Should be before 'javafx.scene.layout.HBox'.

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+import javafx.scene.text.Text;
+
+public class TutorialController {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    private ResourceBundle resources;
+
+    @FXML
+    private URL location;
+
+    @FXML
+    private ImageView imageViewId;
+
+    @FXML
+    private HBox hBoxId;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:25]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.MemberNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Member name 'hBoxId' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$' (camelCase, start with lowercase, only alphanumerical).

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @FXML
+    void endTutorial() {
+        tutorial.startTest();
+    }
+
+    @FXML
+    void initialize() {
+        assert imageViewId != null : "fx:id=\"imageViewID\" was not injected: check your FXML file 'Tutorial.fxml'.";
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:74]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 117).

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+        assert hBoxId != null : "fx:id=\"hBoxId\" was not injected: check your FXML file 'Tutorial.fxml'.";
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:75]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 107).

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+        assert nextButtonId != null : "fx:id=\"nextButtonId\" was not injected: check your FXML file 'Tutorial.fxml'.";
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:76]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 119).

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+        assert backButtonId != null : "fx:id=\"backButtonId\" was not injected: check your FXML file 'Tutorial.fxml'.";
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:77]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 119).

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+        assert progressTextId != null : "fx:id=\"progressTextId\" was not injected: check your FXML file 'Tutorial.fxml'.";
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:78]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 123).

File: src/main/java/github/weichware10/toolbox/gui/TutorialController.java [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+        assert finishButtonId != null : "fx:id=\"finishButtonId\" was not injected: check your FXML file 'Tutorial.fxml'.";
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/TutorialController.java:79]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 123).

**Review by grenoui11e — APPROVED** [review]

**Comment by joshuajeschek** [comment]
Danke für deine Arbeit, ich würde das so anpassen, dass das Tutorial vor dem PreTest gezeigt wird :D

**Comment by joshuajeschek** [comment]
okay wow ich hatte mal wieder vergessen zu pushen xd
@TheDarkDesync kannst du nochmal drüberschauen?

**Review by ghost — APPROVED** [review]
i like it

**Pull request merged**